### DISCUSSION TITLE
关于transform pass的检测

### DISCUSSION THREAD

**Discussion by LunarJune** [arcsysu/SYsU-lang#55] [Q&A]
请问师兄检测是否使用transform pass的方法是查看是否调用<llvm/Transforms>下的接口吗？
我在<llvm/Transforms/Utils/Local.h>中找到了一个方法isInstructionTriviallyDead，用来判断死代码，且不会对IR造成影响，这种可以使用吗？

**Comment by wu-kan** [comment]
不会造成影响就可以，不过实验报告中最好附加不会对ir造成影响的证明或者单独运行这个ir前后对比，以免我误判
